src/sentry/integrations/github/client.py

```diff
@@ -45,7 +47,52 @@ def search_repositories(self, query):
         return self.get("/search/repositories", params={"q": query})
 
     def get_assignees(self, repo):
-        return self.get(f"/repos/{repo}/assignees")
+        return self.get_with_pagination(f"/repos/{repo}/assignees")
+
+    def get_with_pagination(self, path, *args, **kwargs):
+        """
+        Github uses the Link header to provide pagination links. Github recommends using the provided link relations and not constructing our own URL.
+        https://docs.github.com/en/rest/guides/traversing-with-pagination
+        """
+        try:
+            with sentry_sdk.configure_scope() as scope:
+                parent_span_id = scope.span.span_id
+                trace_id = scope.span.trace_id
+        except AttributeError:
+            parent_span_id = None
+            trace_id = None
+
+        with sentry_sdk.start_transaction(
+            op=f"{self.integration_type}.http.pagination",
+            name=f"{self.integration_type}.http_response.pagination.{self.name}",
+            parent_span_id=parent_span_id,
+            trace_id=trace_id,
+            sampled=True,
+        ):
+            output = []
+            resp = self.get(path, params={"per_page": self.page_size})
+            output.extend(resp)
+
+            def get_next_link(resp):
+                link = resp.headers.get("link")
+                if link is None:
+                    return None
+
+                # Should be a comma separated string of links
+                links = link.split(",")
+
+                for link in links:
+                    # If there is a 'next' link return the URL between the angle brackets, or None
+                    if 'rel="next"' in link:
+                        return link[link.find("<") + 1 : link.find(">")]
+
+                return None
+
+            while get_next_link(resp):
```

Comment:
@NisanthanNanthakumar I'm always hesitant to make things have potentially infinite loops. What if Github has a bug where there is always a `link` header? Might be safer to have a max number of pages we'd loop through.

Reply:
@scefali haha yeah I agree. The case is improbable but not impossible. I think we can safely default to 10 requests max, since that'll fetch a total of 1000 records.